File: Data Structures & Algorithms/buy-and-sell-crypto/submission-4.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0, l = 0, r = 1;
        if (prices.size() == 1) return 0;
        while (r < prices.size()) {
            if (prices[l] < prices[r]) max_profit = max(max_profit, prices[r] - prices[l]);
            else l = r;
            r++;
        }
        return max_profit;
    }
};

File: Data Structures & Algorithms/longest-repeating-substring-with-replacement/submission-4.py
class Solution:
    def characterReplacement(self, s: str, k: int) -> int:
        res = 0
        l = 0
        charMap = {}
        for r in range(len(s)):
            charMap[s[r]] = charMap.get(s[r], 0) + 1
            while ((r - l + 1) - max(charMap.values())) > k:
                charMap[s[l]] = charMap.get(s[l], 0) - 1
                l += 1
            res = max(res, r - l + 1)
        return res